Framework/API/src/ExperimentInfo.cpp

```diff
@@ -889,12 +912,13 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
  * this reference.
  */
 const DetectorInfo &ExperimentInfo::detectorInfo() const {
-  if (!m_detectorInfo) {
+  if (!m_detectorInfoWrapper) {
     std::lock_guard<std::mutex> lock{m_detectorInfoMutex};
-    if (!m_detectorInfo)
-      m_detectorInfo = Kernel::make_unique<DetectorInfo>(getInstrument());
+    if (!m_detectorInfoWrapper)
+      m_detectorInfoWrapper =
+          Kernel::make_unique<DetectorInfo>(getInstrument());
   }
-  return *m_detectorInfo;
+  return *m_detectorInfoWrapper;
```

Comment:
This is absolutely intentional. The plan is to step by step move things from `API::DetectorInfo` into `Beamline::DetectorInfo`. The interfaces of the two classes will be kept compatible as far as possible. Once the move is complete this method will be changed to return `Beamline::DetectorInfo`, with little to no changes required in client code.